README.md

```diff
@@ -33,30 +33,44 @@ Prerequisites
     $ cd libtock-c
     ```
 
-1. The main requirement to build the C applications in this repository
-   is a cross compiler for embedded targets. Currently, the only
-   supportetd target is ARM Cortex M, so you will need an
-   `arm-none-eabi` toolchain.
+1. The main requirement to build the C applications in this repository is having
+   cross compilers for embedded targets. You will need an `arm-none-eabi`
+   toolchain for Cortex-M targets, and `riscv64-unknown-elf` toolchain for
+   RISC-V targets.
 
-   MacOS:
+   **MacOS**:
    ```
    $ brew tap ARMmbed/homebrew-formulae && brew update && brew install arm-none-eabi-gcc
+   $ brew tap riscv/riscv && brew update && brew install riscv-gnu-toolchain --with-multilib
    ```
 
-   Ubuntu (18.04LTS or later):
+   You can also download a pre-built RISC-V toolchain
+   [here](http://cs.virginia.edu/~bjc8c/archive/gcc-riscv64-unknown-elf-8.3.0-mac.zip).
+
+   **Ubuntu (18.04LTS or later)**:
    ```
    $ sudo apt install gcc-arm-none-eabi
    ```
 
-   Ubuntu (before 18.04):
+   You will also need a `riscv64` toolchain. You can download a pre-built
+   toolchain
+   [here](http://cs.virginia.edu/~bjc8c/archive/gcc-riscv64-unknown-elf-8.3.0-linux.zip),
+   or follow [these
+   instructions](https://github.com/riscv/riscv-gnu-toolchain#installation-newlib)
```

Comment:
People shouldn't follow these steps. This is an out of date fork that we shouldn't be using.

Reply:
Maybe on a call this week, can you explain this ecosystem? You seem to be in tune with it, but to me it's like black magic. How do you know it is out of date? There is a commit from 20 days ago. Where is the real version? How is anyone supposed to track this?

Reply:
Before RISC-V support was in upstream GCC, binutils, libc's ect. then then RISC-V toolchain fork was the best option.

Now that RISC-V support is supported in all the upstream projects that is what should be used. The upstream projects are heavily tested and shipped in distros. They are reviewed by more people and follow the standard release processes.

The RISC-V fork is pretty much exclusively maintained by SiFive for their them. For example the RISC-V binutils fork description on GitHub says: "RISC-V backports for binutils-gdb. Development is done upstream at the FSF". The fork is not as heavily tested and doesn't have formal release process.

Everyone should be using the upstream projects for toolchains at this point.